### PR TITLE
fix: move etcd to `cri` containerd runner

### LINKED ISSUE #3229: Rebooting Talos in offline mode is not working
## Bug Report
Rebooting a talos node in an air-gapped environment is not working as there is a try to pull Etcd image.

### Description

In reboot of talos node without internet connectivity the reboot hangs. Once restoring the internet connectivity reboot continues. It should be possible to run Talos in air-gapped without local registry. Without internet the reboot is stuck with  `failed to pull image`

Etcd service:
```
  80   // Pull the image and unpack it.
  81   containerdctx := namespaces.WithNamespace(ctx, constants.SystemContainerdNamespace)
  82   if _, err = image.Pull(containerdctx, r.Config().Machine().Registries(), client, r.Config().Cluster().Etcd().Image()); err != nil {
  83     return fmt.Errorf("failed to pull image %q: %w", r.Config().Cluster().Etcd().Image(), err)
  84   }
  85
```

### Environment

- Talos version: 0.8.1
- Kubernetes version: v1.19.1
- Platform: Bare-metal

### PR DESCRIPTION
**Pull request by smira**
This fixes a problem when Talos pulls `etcd` image one every reboot, as
`etcd` was running in the system containerd which is completely
ephemeral (backed by `tmpfs`).

Also skip pulling if image is already present and unpacked (same fix for
the `kubelet` image).

Fixes #3229

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: internal/app/machined/pkg/system/services/kubelet.go [hunk 1]
```diff
@@ -112,9 +114,25 @@ func (k *Kubelet) PreFunc(ctx context.Context, r runtime.Runtime) error {
...
+		if err != nil || !unpacked {
+			pullRequired = true
+		}
+	}
+
+	if pullRequired {
+		_, err = image.Pull(containerdctx, r.Config().Machine().Registries(), client, r.Config().Machine().Kubelet().Image())
+		if err != nil {
+			return err
+		}
```
Comment (andrewrynhard) [inline internal/app/machined/pkg/system/services/kubelet.go:135]: Worth putting this in a common place? Like our `containerd` package.

**Review by andrewrynhard — APPROVED** [review]
I wonder if we should have etcd and kubelet be in the `system` namespace (but still in the CRI containerd). Makes for a cleaner separation. Also, I left a question. I know we need to get just get _something_ so don't let me shoe you down.

File: internal/app/machined/pkg/system/services/kubelet.go (hunk 1 above)
Comment (smira) [reply internal/app/machined/pkg/system/services/kubelet.go:135]: yep, I think that makes sense. probably even `image.Pull` function itself with an option?

**Comment by smira** [comment]
> I wonder if we should have etcd and kubelet be in the `system` namespace (but still in the CRI containerd). Makes for a cleaner separation. Also, I left a question. I know we need to get just get _something_ so don't let me shoe you down.

yeah, we could... `talosctl containers -k` now doesn't see these, as they're not launched via CRI, but we could add even more separation here

File: internal/app/machined/pkg/system/services/kubelet.go (hunk 1 above)
Comment (andrewrynhard) [reply internal/app/machined/pkg/system/services/kubelet.go:135]: Makes sense.

**Comment by smira** [comment]
e

> > I wonder if we should have etcd and kubelet be in the `system` namespace (but still in the CRI containerd). Makes for a cleaner separation. Also, I left a question. I know we need to get just get _something_ so don't let me shoe you down.
> 
> yeah, we could... `talosctl containers -k` now doesn't see these, as they're not launched via CRI, but we could add even more separation here

moved to system namespace and refactored to reduce duplication

**Review by andrewrynhard — APPROVED** [review]

**Pull request merged**